src/transformers/models/mask2former/modeling_mask2former.py

```diff
@@ -0,0 +1,2416 @@
+            multi_stage_positional_embeddings (`List[torch.Tensor]`):
+                positional embeddings for the multi_stage_features
+            attention_mask (`torch.FloatTensor`): attention mask for the masked cross attention layer
+            query_embeddings (`torch.FloatTensor`, *optional*):
+                position embeddings that are added to the queries and keys in the self-attention layer.
+            output_attentions (`bool`, *optional*):
+                Whether or not to return the attentions tensors of all attention layers. See `attentions` under
+                returned tensors for more detail.
+        """
+
+        level_index = index % self.num_feature_levels
+        attention_mask[torch.where(attention_mask.sum(-1) == attention_mask.shape[-1])] = False
+
+        # Masked Cross Attention
+        output, cross_attn_weights = self.cross_attn(
+            output,
+            multi_stage_features[level_index],
+            memory_mask=attention_mask,
+            memory_key_padding_mask=None,  # here we do not apply masking on padded region
+            pos=multi_stage_positional_embeddings[level_index],
+            query_pos=query_embeddings,
+        )
+
+        # Self Attention
+        output, self_attn_weights = self.self_attn(
+            output,
+            output_mask=None,
+            output_key_padding_mask=None,
+            query_pos=query_embeddings,
+        )
+
+        # Fully Connected
+        output = self.ffn(output)
+
+        outputs = (output,)
+
+        if output_attentions:
+            outputs += (self_attn_weights, cross_attn_weights)
+
+        return outputs
+
+
+class Mask2FormerTransformerDecoder(nn.Module):
+    """
+    Transformer decoder
+    """
+
+    def __init__(self, in_channels: int, config: Mask2FormerConfig):
+        super().__init__()
+        self.config = config
+        self.num_heads = config.decoder_config["num_heads"]
+        self.is_train = config.general_config["is_train"]
+        self.use_auxiliary_loss = config.general_config["use_auxiliary_loss"]
+
+        self.decoder_norm = nn.LayerNorm(config.decoder_config["hidden_dim"])
+        self.num_feature_levels = 3
+
+        self.layers = nn.ModuleList(
+            [Mask2FormerTransformerDecoderLayer(config) for _ in range(config.decoder_config["decoder_layers"] - 1)]
+        )
+
+        self.class_embed = nn.Linear(config.decoder_config["hidden_dim"], config.general_config["num_classes"] + 1)
+        self.mask_embed = Mask2FormerMLPPredictionHead(
+            config.decoder_config["hidden_dim"],
+            config.decoder_config["hidden_dim"],
+            config.decoder_config["mask_feature_size"],
+            3,
+        )
+
+    def forward(
+        self,
+        multi_stage_features=None,
+        multi_stage_positional_embeddings=None,
+        mask_features=None,
+        output=None,
+        query_embeddings=None,
+        size_list=None,
+        output_attentions=None,
```

Comment:
Would be great to be aligned here with the other implementations (DETR, Deformable DETR, MaskFormer) in terms of API; meaning

* support for both `output_hidden_states` and `output_attentions`
* same termininology e.g. it seems output corresponds to the `hidden_states` (object query embeddings) and use this as first argument